packages/client/lib/rpc/modules/eth.ts

```diff
@@ -986,4 +988,55 @@ export class Eth {
     const block = await getBlockByOption(blockOpt, this._chain)
     return intToHex(block.transactions.length)
   }
+
+  /**
+   * Gas price oracle.
+   *
+   * Returns a suggested gas price.
+   * @returns a hex code of an integer representing the current gas price in wei.
+   */
+  async gasPrice() {
+    let gasPrice = BigInt(0)
+    const latest = await this._chain.getCanonicalHeadHeader()
+    if (this._vm !== undefined && this._vm._common.isActivatedEIP(1559)) {
+      const baseFee = latest.calcNextBaseFee()
+      let priorityFee = BigInt(0)
+      const block = await this._chain.getBlock(latest.number)
+      for (const tx of block.transactions) {
+        const maxPriorityFeePerGas = (tx as FeeMarketEIP1559Transaction).maxPriorityFeePerGas
+        priorityFee += maxPriorityFeePerGas
+      }
+
+      priorityFee =
+        priorityFee !== BigInt(0) ? priorityFee / BigInt(block.transactions.length) : BigInt(1)
+      gasPrice = baseFee + priorityFee
+    } else {
+      // For chains that don't support EIP-1559 we iterate over the last 20
+      // blocks to get an average gas price. We cap the total tx lookup to 500.
+      const blockIterations = 20 < latest.number ? 20 : latest.number
+      let txCount = 0
+      for (let i = 0; i < blockIterations; i++) {
+        const block = await this._chain.getBlock(latest.number - BigInt(i))
+        if (block.transactions.length === 0) {
+          continue
+        }
+
+        for (const tx of block.transactions) {
+          const txGasPrice = (tx as Transaction).gasPrice
+          gasPrice += txGasPrice
+          txCount++
+          if (txCount >= 500) break
```

Comment:
I really do not think we should do a per-block check, and I also do not think we need a tx cap limit. I might be wrong here, but the heaviest operation here is pulling the block body from the DB. Iterating over the txs should be much faster than that.

Reply:
Yes this makes sense. I removed the tx cap.